build-tools/packages/build-cli/src/commands/publish/tarballs.ts

```diff
@@ -0,0 +1,226 @@
+			default: ".",
+		}),
+		orderFile: Flags.file({
+			description:
+				"A file with package names that should be published. Such files can be created using `flub list`.",
+			exists: true,
+			required: true,
+		}),
+		tarball: Flags.boolean({
+			description:
+				"Use this flag to indicate that the orderFile contains tarball names instead of package names. Such files can be created using `flub list --tarball`. This option is deprecated and for backwards compatibility only.",
+			dependsOn: ["orderFile"],
+			deprecated: {
+				message: "This option is deprecated and for backwards compatibility only.",
+			},
+		}),
+		retry: Flags.integer({
+			description: `Number of times to retry publishing a package that fails to publish.`,
+			default: 0,
+		}),
+		dryRun: Flags.boolean({
+			aliases: ["dry-run"],
+			description:
+				"Does everything except publish to the registry. This flag will be passed to 'npm publish'.",
+			default: false,
+		}),
+		access: Flags.string({
+			description: "This flag will be passed to 'npm publish'.",
+			options: ["public", "restricted"],
+		}),
+		publishArgs: Flags.string({
+			description:
+				"This string will be passed to 'npm publish' verbatim. Use this to pass additional custom args to npm publish like --tag.",
+		}),
+	};
+
+	public async run(): Promise<void> {
+		const {
+			access,
+			dir,
+			dryRun,
+			orderFile,
+			retry,
+			tarball: orderFileIsTarballs,
+			publishArgs: rawPublishArgs,
+		} = this.flags;
+		const publishArgs: string[] = (rawPublishArgs ?? "").split(" ");
+		if (access !== undefined) {
+			publishArgs.push("--access", access);
+		}
+		if (dryRun) {
+			publishArgs.push("--dry-run");
+		}
+
+		// eslint-disable-next-line unicorn/no-await-expression-member
+		const packageOrder = (await readLines(orderFile)).filter((line) => {
+			return line !== "" && line !== undefined;
+		});
+
+		const tarballs = await globby(["*.tgz"], { cwd: dir, absolute: true });
+		const tarballMetadata = new Map<string, TarballMetadata>();
+
+		const mapPromises: Promise<void>[] = [];
+		for (const tarballPath of tarballs) {
+			mapPromises.push(
+				extractPackageJsonFromTarball(tarballPath).then((json) => {
+					const tarballName = getTarballName(json);
+					tarballMetadata.set(tarballName, {
+						name: json.name,
+						version: json.version,
+						filePath: tarballPath,
+						fileName: path.basename(tarballPath),
+					});
+				}),
+			);
+		}
+		await Promise.all(mapPromises);
+
+		for (const entry of packageOrder) {
+			const lookupEntry = orderFileIsTarballs ? entry : getTarballName(entry);
+			const toPublish = tarballMetadata.get(lookupEntry);
+			if (toPublish === undefined) {
+				this.error(`No tarball found matching '${entry}'`, { exit: 1 });
+			}
+
+			let tryCount = 0;
+			let status: PublishStatus;
+
+			do {
+				this.info(`Publishing ${toPublish.fileName}, attempt ${tryCount + 1}`);
+				// eslint-disable-next-line no-await-in-loop -- This should execute serially
```

Comment:
I believe this is because of the packageOrder rule (even if some could actually be done in parallel). Can you update the comment to be more explicit about the order requirement and perhaps state why (which I can guess at :) )